plugin/trino-kinesis/src/main/java/io/trino/plugin/kinesis/KinesisRecordSet.java

```diff
@@ -290,6 +300,32 @@ private void getKinesisRecords()
             messagesRead += kinesisRecords.size();
         }
 
+        private Optional<Map<DecoderColumnHandle, FieldValueProvider>> decodeMessage(byte[] messageData)
+        {
+            if (!split.getCompressionCodec().isPresent()) {
+                return messageDecoder.decodeRow(messageData);
+            }
+            if (split.getCompressionCodec().get() == GZIP || split.getCompressionCodec().get() == AUTOMATIC) {
+                if (isGZipped(messageData)) {
```

Comment:
Please invert the condition, so there will be less code nested, and corner case will be handled first.

Reply:
The idea was to skip testing the GZIP header if it's not required